permissions/permission_definitions.go

```diff
@@ -0,0 +1,699 @@
+		},
+		{
+			field:             "repo_path",
+			objectType:        "repo",
+			requestObjectType: "repos",
+			allowedPermissionLevels: map[string]permissionLevelOptions{
+				"CAN_READ":   {isManagementPermission: false},
+				"CAN_RUN":    {isManagementPermission: false},
+				"CAN_EDIT":   {isManagementPermission: false},
+				"CAN_MANAGE": {isManagementPermission: true},
+			},
+			idRetriever:        PATH,
+			readAclCustomizers: []aclReadCustomizer{removeAdminPermissionsCustomizer},
+		},
+		{
+			field:             "authorization",
+			objectType:        "tokens",
+			requestObjectType: "authorization",
+			stateMatcher: func(id string) bool {
+				return id == "tokens"
+			},
+			idMatcher: func(id string) bool {
+				return id == "/authorization/tokens"
+			},
+			allowedPermissionLevels: map[string]permissionLevelOptions{
+				"CAN_USE":    {isManagementPermission: true},
+				"CAN_MANAGE": {isManagementPermission: true},
+			},
+			updateAclCustomizers: []aclUpdateCustomizer{
+				addAdminAclCustomizer(func(objectId string) bool {
+					return objectId == "/authorization/tokens"
+				}),
+			},
+		},
+		{
+			field:             "authorization",
+			objectType:        "passwords",
+			requestObjectType: "authorization",
+			stateMatcher: func(id string) bool {
+				return id == "passwords"
+			},
+			idMatcher: func(id string) bool {
+				return id == "/authorization/passwords"
+			},
+			allowedPermissionLevels: map[string]permissionLevelOptions{
+				"CAN_USE": {isManagementPermission: true},
+			},
+		},
+		{
+			field:             "sql_endpoint_id",
+			objectType:        "warehouses",
+			requestObjectType: "sql/warehouses",
+			allowedPermissionLevels: map[string]permissionLevelOptions{
+				"CAN_USE":     {isManagementPermission: false},
+				"CAN_MANAGE":  {isManagementPermission: true},
+				"CAN_MONITOR": {isManagementPermission: false},
+				"IS_OWNER":    {isManagementPermission: true},
+			},
+			updateAclCustomizers: []aclUpdateCustomizer{addCurrentUserAsManageCustomizer},
+			deleteAclCustomizers: []aclUpdateCustomizer{addCurrentUserAsManageCustomizer},
+			readAclCustomizers:   []aclReadCustomizer{removeAdminPermissionsCustomizer},
+			fetchObjectCreator: func(ctx context.Context, w *databricks.WorkspaceClient, objectID string) (string, error) {
+				warehouse, err := w.Warehouses.GetById(ctx, strings.ReplaceAll(objectID, "/sql/warehouses/", ""))
+				if err != nil {
+					return "", common.IgnoreNotFoundError(err)
+				}
+				return warehouse.CreatorName, nil
+			},
+		},
+		{
+			field:             "sql_dashboard_id",
+			objectType:        "dashboard",
+			requestObjectType: "dbsql-dashboards",
+			allowedPermissionLevels: map[string]permissionLevelOptions{
+				"CAN_EDIT":   {isManagementPermission: false},
+				"CAN_RUN":    {isManagementPermission: false},
+				"CAN_MANAGE": {isManagementPermission: true},
+				// This was originally called CAN_VIEW in the preview API, but was renamed to CAN_READ in the GA API.
```

Comment:
we are blocking `CAN_READ`, which is a valid configuration here?

Reply:
At first, I wasn't sure we would be able to support it, because on the read pathway we would need to convert CAN_READ to CAN_VIEW when users used CAN_VIEW. I've made some tweaks and think it should be possible to support both now since the read pathway has access to the current state.